app/src/main/java/org/oppia/app/parser/StringToNumberWithUnitsParser.kt

```diff
@@ -0,0 +1,168 @@
+package org.oppia.app.parser
+
+import org.oppia.app.model.CurrencyUnit
+import org.oppia.app.model.Fraction
+import org.oppia.app.model.NumberUnit
+import org.oppia.app.model.NumberWithUnits
+import org.oppia.app.model.Units
+import org.oppia.domain.util.normalizeWhitespace
+import java.lang.Double
+
+import java.util.regex.Pattern
+
+/** This class contains methods that help to parse string to number, check realtime and submit time errors. */
```

Comment:
Reminder to add link to Oppia web version of this logic for reference.

Reply:
Done.